tools/generate-function-bindings/scripts/generate.ts

```diff
@@ -0,0 +1,415 @@
+        if (type[1] === "string") {
+          child.mapper = "anyValue";
+        }
+
+        return {
+          mapper: `listOf(${child.mapper})`,
+          tsType: t.tsArrayType(child.tsType),
+          docstringType: `Array<${child.docstringType}>`,
+        };
+      } else if (Array.isArray(type) && type[0] === "map") {
+        const child = parseType(type[1]);
+        return {
+          mapper: "mapValue",
+          tsType: t.tsAnyKeyword(),
+          docstringType: "Object<string, " + child.docstringType + ">",
+        };
+      } else {
+        throw new Error(
+          `Function ${name} has parameter ${
+            p.name
+          } with unsupported type ${JSON.stringify(p.type)}`
+        );
+      }
+    };
+
+    const { docstringType, mapper, tsType } = parseType(p.type);
+
+    const tsParam = t.identifier(name);
+    tsParam.typeAnnotation = t.tsTypeAnnotation(tsType);
+
+    return { name, mapper, tsParam, docstringType };
+  };
+
+  const parameters: MappedParameter[] = (signature.parameters || []).map(
+    mapParameter
+  );
+
+  if (signature.variadic_parameter) {
+    const p = mapParameter(signature.variadic_parameter);
+    p.tsParam.typeAnnotation = t.tsTypeAnnotation(
+      t.tsArrayType(
+        (p.tsParam.typeAnnotation as t.TSTypeAnnotation).typeAnnotation
+      )
+    );
+    parameters.push({
+      name: p.name,
+      docstringType: `Array<${p.docstringType}>`,
+      mapper: `variadic(${p.mapper})`,
+      tsParam: p.tsParam,
+    });
+  }
+
+  // we need a space (Prettier will remove it) as somehow ts`` works in weird ways when
+  // passing an empty (or falsy value in the template string)
+  const argValueMappers: string =
+    parameters.map((p) => p.mapper).join(",") || " ";
+  const argNames: string = parameters.map((p) => p.name).join(",");
+  const params: any[] = parameters.map((p) => p.tsParam);
+
+  const body = ts`
+  return ${returnType}(terraformFunction("${name}", [${argValueMappers}])(${argNames}));
+  `();
+
+  const isInternal = INTERNAL_METHODS.includes(name);
+
+  let sanitizedFunctionName = name === "length" ? "lengthOf" : name;
+  if (isInternal) {
+    sanitizedFunctionName = `_${sanitizedFunctionName}`;
+  }
+
+  const method = t.classMethod(
+    "method",
+    t.stringLiteral(sanitizedFunctionName),
+    params,
+    t.blockStatement(Array.isArray(body) ? body : [body]),
+    false, // computed
+    true // static
+  );
+
+  // comment with docstring for method
+  const descriptionWithLink = signature.description.replace(
+    `\`${name}\``,
+    `{@link https://developer.hashicorp.com/terraform/language/functions/${name} ${name}}`
+  );
+  t.addComment(
+    method,
+    "leading",
+    [
+      "*",
+      ...(isInternal ? ["* @internal"] : []),
+      `* ${descriptionWithLink}`,
+      ...parameters.map((p) => ` * @param {${p.docstringType}} ${p.name}`),
+      "",
+    ].join("\n")
+  );
+
+  return method;
+}
+
+fetchMetadata();
+
+const old = [
```

Comment:
Can't we get rid of the old order now?

Reply:
I opted to keep it for now, in case someone wanted to compare the generated file from this PR to our terraform-functions file. This way, a reviewer doesn't need to search for the commit which had the sorted generated file in it.
Shall I create an follow-up item or just remove it now? No strong opinion here.

Reply:
removed 👍